### DISCUSSION TITLE
is there a way to deserialize a multidocument yaml

### DISCUSSION THREAD

**Discussion by camerondavison** [charleskorn/kaml#293] [Q&A]
Snake YAML has a `loadAll` I was curious if there is way to get something like

```
---
firstName: "John"
lastName: "Doe"
age: 20
---
firstName: "Jack"
lastName: "Jones"
age: 25
```

to work. right now it seems like it is getting
```
 com.charleskorn.kaml.MalformedYamlException at <root> on line 10, column 0: Unexpected DocumentStart, expected StreamEnd
```

**Comment by charleskorn** [comment]
kaml does not support decoding multiple documents.

How would you expect this to be translated to Kotlin types?

**Reply by camerondavison** [reply]
I think it would be an iterator of a kotlin type.

Where from each start to end would be 1 object.

**Reply by charleskorn** [reply]
Seems reasonable to me.

Would you be interested in submitting a PR to implement this?

**Reply by camerondavison** [reply]
I could try. But I don't have a bunch of time. You would be okay adding a new set of methods that are named deserializeAllStream or something?

**Reply by charleskorn** [reply]
I'm imagining something like these two functions:

```kotlin
fun <T> decodeDocumentsFromString(deserializer: DeserializationStrategy<T>, string: String): List<T>
fun <T> encodeDocumentsToString(serializer: SerializationStrategy<T>, documents: List<T>): String
```

How does that sound?
